### PR TITLE
:tada: Allow user to trigger processes from the process detail view, for MANUAL trigger

### PR DESCRIPTION
**Pull request by mdartic**
This PR should allow users to trigger directly from the `ProcessDetail` view a `MANUAL` process, related to a workspace and not a table directly.

### CODE REVIEW

File: front/src/services/lck-api/definitions.ts [hunk 1]
```diff
@@ -372,6 +372,14 @@ export class MediaConfiguration {
   type?: MEDIA_TYPE.IMAGE | MEDIA_TYPE.VIDEO;
 }
 
+export enum PROCESS_TRIGGER {
+  CREATE_ROW = 'CREATE_ROW', // when a row in inserted
+  UPDATE_ROW = 'UPDATE_ROW', // when a row is updated, no matter which data
+  UPDATE_ROW_DATA = 'UPDATE_ROW_DATA', // when a data in a row is updated
+  CRON = 'CRON',
```
Comment (Alketta) [inline front/src/services/lck-api/definitions.ts:379]: Finally, we keep `CRON` ?

File: front/src/components/store/Process/ProcessForm.vue [hunk 2]
```diff
@@ -64,57 +66,68 @@
...
+        field="text"
+        :suggestions="tables"
+        @search="$emit('search-table', $event)"
+        v-model="processCloned.table"
+        @item-select="processCloned.table_id = processCloned.table.value"
+        @clear="processCloned.table_id = null"
       />
+      <small
+        class="p-text-italic"
+        id="password-rules"
```
Comment (Alketta) [inline front/src/components/store/Process/ProcessForm.vue:124]: ?

File: api/src/services/row/triggerProcess.test.ts [hunk 3]
```diff
@@ -93,11 +93,15 @@ describe('\'triggerProcess\' hook', () => {
     expect.assertions(1)
 
     const processTriggerManual = await app.service('process').create({
+      workspace_id: workspace.id,
+      url: 'http://localhost',
       table_id: table1.id,
       trigger: ProcessTrigger.MANUAL,
       enabled: true,
     }) as Process
     const processTriggerCron = await app.service('process').create({
```
Comment (Alketta) [inline api/src/services/row/triggerProcess.test.ts:102]:
```suggestion
    const processTriggerManual = await app.service('process').create({
```

**Review by Alketta — COMMENTED** [review]
Maybe we need to add more tests to check 
 1. An user with aclset manager cannot create a process in another workspace
 2. An user without aclset manager cannot create a process in this workspace
 3. An user without aclset manager can trigger a process in this workspace

File: api/src/abilities/process.abilities.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+          $joinRelation: '[group.[aclset]]',
+          'group:aclset.workspace_id': currentWorkspaceId,
+          'group:aclset.manager': true,
+        },
+        paginate: false,
+      }) as Usergroup[]
+      if (usergroups.length === 0) {
+        throw new Forbidden('You do not have access to this workspace with enough privilege. Please ask a workspace manager to join this workspace with a manager role.')
+      }
+      can('manage', 'process', {
```
[the remark this answers is not in the document]
Comment (mdartic) [reply api/src/abilities/process.abilities.ts:87]: Yup, but, that's already the case ?
If he tries to crud a process on another workspace, the abilities will be computed for this workspace ?

File: api/src/abilities/process_run.abilities.ts [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+      /**
+       * Check the user have access to this workspace
+       */
+      const usergroups = await services.usergroup._find({
+        query: {
+          user_id: user.id,
+          $eager: '[group.[aclset.[acltables]]]',
+          $joinRelation: '[group.[aclset]]',
+          'group:aclset.workspace_id': currentWorkspaceId,
+          'group:aclset.manager': true,
```
[the remark this answers is not in the document]
Comment (mdartic) [reply api/src/abilities/process_run.abilities.ts:85]: Well, I think you're right.
I'll check that.

File: front/src/components/store/Process/ProcessForm.vue (hunk 2 above)
Comment (mdartic) [reply front/src/components/store/Process/ProcessForm.vue:124]: ! Bad copy paste.

File: front/src/services/lck-api/definitions.ts (hunk 1 above)
Comment (mdartic) [reply front/src/services/lck-api/definitions.ts:379]: No... bad sync between front & back...

**Comment by mdartic** [comment]
Well in fact tests you want are already here in the code :-)
see https://github.com/locokit/locokit/pull/116/files#diff-151c547ea3ad4c8c28be3c0a41e945dd40604bf650df38528eded518d677974aR298-R408

**Review by Alketta — APPROVED** [review]

**Pull request merged**